cpp/tests/centrality/edge_betweenness_centrality_test.cu

```diff
@@ -0,0 +1,340 @@
+// SSSP's test suite code uses type of Graph parameter that could be used
+// (MTX / RMAT)
+// FIXME: Use VT for number_of_sources?
+typedef struct EdgeBC_Usecase_t {
+  std::string config_;     // Path to graph file
+  std::string file_path_;  // Complete path to graph using dataset_root_dir
+  int number_of_sources_;  // Starting point from the traversal
+  EdgeBC_Usecase_t(const std::string &config, int number_of_sources)
+    : config_(config), number_of_sources_(number_of_sources)
+  {
+    // assume relative paths are relative to RAPIDS_DATASET_ROOT_DIR
+    // FIXME: Use platform independent stuff from c++14/17 on compiler update
+    const std::string &rapidsDatasetRootDir = get_rapids_dataset_root_dir();
+    if ((config_ != "") && (config_[0] != '/')) {
+      file_path_ = rapidsDatasetRootDir + "/" + config_;
+    } else {
+      file_path_ = config_;
+    }
+  };
+} EdgeBC_Usecase;
+
+class Tests_EdgeBC : public ::testing::TestWithParam<EdgeBC_Usecase> {
+ public:
+  Tests_EdgeBC() {}
+  static void SetupTestCase() {}
+  static void TearDownTestCase() {}
+
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+  // FIXME: Should normalize be part of the configuration instead?
+  // VT         vertex identifier data type
+  // ET         edge identifier data type
+  // WT         edge weight data type
+  // result_t   result data type
+  // normalize  should the result be normalized
+  template <typename VT, typename ET, typename WT, typename result_t, bool normalize>
+  void run_current_test(const EdgeBC_Usecase &configuration)
+  {
+    // Step 1: Construction of the graph based on configuration
+    bool is_directed = false;
+    auto csr = generate_graph_csr_from_mm<VT, ET, WT>(is_directed, configuration.file_path_);
+    cudaDeviceSynchronize();
+    cugraph::experimental::GraphCSRView<VT, ET, WT> G = csr->view();
+    G.prop.directed                                   = is_directed;
+    CUDA_CHECK_LAST();
+    std::vector<result_t> result(G.number_of_edges, 0);
+    std::vector<result_t> expected(G.number_of_edges, 0);
+
+    // Step 2: Generation of sources based on configuration
+    //         if number_of_sources_ is 0 then sources must be nullptr
+    //         Otherwise we only  use the first k values
+    ASSERT_TRUE(configuration.number_of_sources_ >= 0 &&
+                configuration.number_of_sources_ <= G.number_of_vertices)
+      << "Number number of sources should be >= 0 and"
+      << " less than the number of vertices in the graph";
+    std::vector<VT> sources(configuration.number_of_sources_);
+    thrust::sequence(thrust::host, sources.begin(), sources.end(), 0);
+
+    VT *sources_ptr = nullptr;
+    if (configuration.number_of_sources_ > 0) { sources_ptr = sources.data(); }
+
+    reference_edge_betweenness_centrality(G,
+                                          expected.data(),
+                                          normalize,
+                                          // FIXME: weights
```

Comment:
remove. it is mentioned a few times at better spots already.

Reply:
Updated in 7e7b668